lib/dialects/mssql/query-interface.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+/**
+ Returns an object that treats MSSQL's inabilities to do certain queries.
+
+ @class QueryInterface
+ @static
+ */
+
+/**
+  A wrapper that fixes MSSQL's inability to cleanly remove columns from existing tables if they have a default constraint.
+
+  @method removeColumn
+  @for    QueryInterface
+
+  @param  {String} tableName     The name of the table.
+  @param  {String} attributeName The name of the attribute that we want to remove.
+  @param  {Object} options
+  @param  {Boolean|Function} [options.logging] A function that logs the sql queries, or false for explicitly not logging these queries
+ */
+var removeColumn = function (tableName, attributeName, options) {
+  var self = this;
+  options = options || {};
+
+  var findConstraintSql = self.QueryGenerator.getDefaultConstraintQuery(tableName, attributeName);
+  return self.sequelize.query(findConstraintSql + ';', { raw: true, logging: options.logging})
+    .spread(function (results) {
+      if (!results.length) {
+          // No default constraint found -- we can cleanly remove the column
+          return;
+      }
+      var dropConstraintSql = self.QueryGenerator.dropConstraintQuery(tableName, results[0].name);
+      return self.sequelize.query(dropConstraintSql + ';', { raw: true, logging: options.logging});
```

Comment:
Not really -- initially I thought this was the standard approach, but I can see that it varies quite a lot! Have moved the semicolons to query-generator.